backend/schema/corpora_schema.md

```diff
@@ -0,0 +1,323 @@
+
+## Schema
+
+### Original Dataset Requirements
+
+
+These are the fundamental requirements that all dataset must meet, the handful of things our software depends on and mandatory legal requirements. All
+submitted data must comply with these requirements.
+
+
+*   **Unique observation identifiers**. Each observation (usually a cell) must have an id that is unique within the dataset. This is usually already
+    present in every submission, for example as a barcode.
+*   **Unique feature identifiers**. Every feature (usually a gene or transcript) also needs a unique identifier. This is occasionally not present because
+    of one-to-many mappings between gene symbols and other gene ids. In cases where there are duplicated feature identifiers, they will be summed during
+    curation.
+*   **Type annotations or inferrable types**. For both expression values and metadata, we don't want to have to guess at the type. Typing is already explicit
+    in anndata, loom, Seurat and Bioconductor formats.
+*   **No PII**. No metadata can be personally identifiable: no names, dates of birth, specific locations, etc. There’s a
+    [list](https://docs.google.com/document/d/1nlUuRiQ8Awo_QsiVFi8OWdhGONjeaXE32z5GblVCfcI/edit?usp=sharing).
+
+
+#### Matrix Layers
+
+The count matrix itself can exist in several forms. Generally, there is a “raw” count matrix without scaling, filtering, normalization, etc. And there is
+a “final” matrix that is used in the publication with QC and various corrections. There can also be several intermediate matrices.
+
+These different transformations of the count matrix are called “layers” in several file formats. Submissions to Corpora must contain at least one layer,
+the “raw” layer. They may optionally also have a “final” layer used for presentation, and each layer must be identified.
+
+
+### Remix Metadata
+
+The remaining part of the schema defines what’s needed for the Corpora remix. The words “must” and “requirement” are used throughout, but note the
+discussion above regarding curation policies as well the [caution against](#Software-vs-policy) introducing any
+software dependencies on these fields.
+
+
+#### Integration Metadata
+
+To support data integration, each cell must have the following metadata:
+
+**Field name**|**Constraints**
+:--|:--
+tissue|string
+assay|string
+disease|string
+cell\_type|string
+sex|“male”, “female”, “unknown”, or “na”
+ethnicity|string, “na” if non-human
+development\_stage|string
+
+In addition to these free text fields (except sex), each cell must also have ontology values:
+
+**Field name**|**Constraints**
+:--|:--
+tissue\_ontology|UBERON term
+assay\_ontology|EFO term
+disease\_ontology|MONDO term or [PATO:0000461](http://bioportal.bioontology.org/ontologies/PATO?p=classes&conceptid=PATO%3A0000461)
+cell\_type\_ontology|CL term
+ethnicity\_ontology|HANCESTRO term, “na” if non-human
+development\_stage\_ontology|HsapDv term if human, child of EFO:0000399 otherwise
```

Comment:
for terms which may come from multiple ontologies, it might be more explicit to allow the encoding of the actual ontology as well as the ontological term ID.   I would make this optional, with reasonable defaults (eg, assume UBERON for tissue, etc).

In addition, given how fluid ontologies are, do you want to consider the ability to specify the ontology version explicitly?

Reply:
For "term" I pretty much always mean "compact uri", so it would be like "HsapDv:00001". And you can reliably get the ontology by splitting on `:`. I'll clarify that.

As far a version, most of the ontologies seem pretty weakly versioned. There are release dates, but those can be pretty frequent, and they don't do much to show whether one term has changed.

Reply:
got it - I didn't understand that you were using this term format.  Definitely should specify that.  

Also suggest more prescriptive language around what is required, vs suggested, so we can draw a straight line from the spec to a validator (software)